### ISSUE TITLE
Allow support for AWS SSL Certificates

### ISSUE THREAD

**Issue by tomtom87** [zodern/meteor-up#565]
tl:dr; Please enable HTTPS protocol when there is no SSL Cert or key file given, for load balancer support.

`mup` is broken for AWS users that are using the Elastic Load Balancer and DNS at Route53 (Amazons DNS provider) which in turn provides SSL certificates. This was working in previous versions of `mupx` but in this latest `mup` SSL from AWS is not supported as it requires a CRT file.

There is no file provided by AWS, simply the SSL cert is on the load balancer and the App must be accessed via HTTPS protocol. 

This was working fine with the `force-ssl` package from atmosphere, but in the latest version we cannot deploy with just setting https the nginx gives a 503 gateway errror in this configuration.

Please fix so we have the option of just allowing SSL via protocol specification of the env var

**Comment by bennygenel** [comment]
@tomtom87 Hi! I'm planning to move to AWS but don't know about how to use ELB. Can I contact you with email or something. I would really appreciate if you help me.

Thanks

**Comment by tomtom87** [comment]
@bennygenel yes sure thing, feel free to give me a private message about that. I hope we can get the SSL working with this soon

**Comment by bennygenel** [comment]
@tomtom87 couldn't figure out how to private message. Could you please send me an email on bennygenel gmail

**Comment by tomtom87** [comment]
@bennygenel  have emailed you about AWS

**Comment by zodern** [comment]
@tomtom87 I want to make sure I understand your situation: ELB is handling ssl, and you want to use the force-ssl pacakge. Is that correct?

It should be possible. The warning about using `force-ssl` does not apply for you since ELB is handling ssl, so you can safely ignore it (I will remove the warning in the next version). If you remove `meteor.ssl` from your config, you will no longer have the 503 error.

**Comment by tomtom87** [comment]
> @tomtom87 I want to make sure I understand your situation: ELB is handling ssl, and you want to use the force-ssl pacakge. Is that correct?

@zodern exactly :)

I am still getting the error without the meteor.ssl part of the config though. Please let me know when you update, very thankful for this mate

**Comment by santteegt** [comment]
@tomtom87 can I also send you an email asking for some help on how to set up SSL on my EC2 instance using mup for meteor deployment ?

**Comment by idanwe** [comment]
@santteegt @tomtom87 can you post here the transition to AWS and the setting of the load balancer with mup. 
I will be my next tasks this week and would really appreciate some references.

**Comment by tomtom87** [comment]
@idanwe  @santteegt  Hi guys, did you manage to get it working on AWS? It's really pretty long winded.

You have to setup an EB load balancer then deploy to the instance, as long as your DNS is at the AWS Route 53 you can use their provided SSL. Still I am unable to get this to work with Meteor in the new version sadly. 

Worked perfect before... wish it still did

**Comment by avishaan** [comment]
Has anyone had luck deploying mup where the load balancer has a valid certificate but deploying to mup without a certificate. 

I want to then add a CNAME record to the load balancer to get something like `beta.company.com` so that when a user goes to beta.company.com it goes to the load balancer. I have everything setup AFAIK such that when I go to the load balancer address it seems to be working with the secure connection but when I go to `beta.company.com` it says 
![pasted_image_7_8_17__4_42_pm](https://user-images.githubusercontent.com/5360982/27988958-77c7c174-63fc-11e7-9da3-0c619f9f227a.png)

**Comment by tomtom87** [comment]
I also had similar errors when trying with a beta screening url - ended up resorting to using a nginx setup for that. Couldnt get mupx working...

**Comment by elie222** [comment]
Any progress here?

**Comment by tomtom87** [comment]
@elie222 it's only working in the old version, the latest one is broken. Just down grade everything to back in May, you are forced to use the SSL provided by AWS and in-turn their DNS which is a dependency of using the SSL. That was the only way I could get mup working

**Comment by LevanArabuli** [comment]
Still no updates? 

@tomtom87 have you been able to find a workaround?

I'm setting up ELB with ACM issued cert, Meteor app is deployed on EC2. Want to remove ssl/proxy dependencies from `mup.js` completely and let ELB run it.

**Comment by LevanArabuli** [comment]
Resolved, classic ELB doesn't support WebSockets/DDP. Use new Application Load Balancer type instead. 
For more details check #914.

**Comment by thirtyfoursoft** [comment]
Hi @LevanArabuli , 

I am using the Application load balancer, but it is also prividing the SSL for https, and I am also using letsencrypt for my app now it is crashing the app. See #972

**Comment by tomtom87** [comment]
@thirtyfoursoft try using the SSL from Route 53 mate

**Comment by sontn** [comment]
> @thirtyfoursoft try using the SSL from Route 53 mate

Hi Tom, can I have your email address? Thank you!
